flexget/plugins/input/anilist.py

```diff
@@ -142,14 +144,45 @@ def on_task_input(self, task, config):
                                 and anime.get('format').lower() in selected_formats
                             )
                         ) or 'all' in selected_formats
+
                         if has_selected_type and has_selected_release_status:
+                            try:
+                                ids = task.requests.post(
+                                    'https://relations.yuna.moe/api/ids',
+                                    json={'anilist': anime.get('id')},
+                                ).json()
+                            except RequestException as e:
+                                ids = {}
+                                raise plugin.PluginWarning(f'Couldn\'t fetch additional IDs - {e}')
+
                             entry = Entry()
+                            entry['al_id'] = anime.get('id') or ids.get('anilist')
```

Comment:
Generally it's better to do
```
anime.get('id', ids.get('anilist'))
```

because using or depends how return value of "id" evaluates in boolean context, not if the field is present or not

Reply:
Well, I did get scolded once in another project for nesting and chaining `get()` so I "learned" to favor the `or` syntax 😅

TBF this fallback isn't even necessary, `id` is a required field (possibly the only one) as it's the index. This `or` ends up more like a documentation that the field is available from that second API due to the short-circuit, so I could just comment it instead.
`idMAL` on the other hand might not be present, but if it is the value is also an integer

Reply:
Hahaa, yeah. I am always a bit torn on this subject. But I think the nesting is correct way as using or does not work as expected if value is 0 or "" for example.

Reply:
Not completely unexpected, both `0` and `""` would be invalid after all xD